[Jungbae Kim] iP

jbkim1999 opened the pull request:
# ISTJBot.
## It is a chat-bot.
### A good chat-bot.
> This bot is clearly of ISTJ MBTI type... - JB Kim
- This bot has many commands.
- But is simple to use.
- Below is a main class.
```java
    public static void main(String[] args) {
        new IstjBot("data/tasks.txt").run();
    }
```
View my cloned repo [here!](https://github.com/jbkim1999/ip)
1. Use main class the launch the bot.
2. Then type in commands.
3. The bot will respond to you!
- [X] Finished Week 3 Tasks
- [ ] Can add more JUnit tests
- [ ] Add headings to each class, then fix 🧐 minor syntax errors

`run` method will run once the `main` method is called from CLI.

**Have fun!**
*Thank you for checking my codes*
~~It's not that perfect...~~

Commit:
Add Gradle support

jbkim1999 committed:
Implemented Level-1

jbkim1999 committed:
Implemented Level-2

jbkim1999 committed:
Implemented Level-3

jbkim1999 committed:
Implemented Level-4. But code not yet refactored

jbkim1999 committed:
A-TextUiTesting implemented

jbkim1999 committed:
Implemented Level-5

jbkim1999 committed:
Implemented Level-6

jbkim1999 committed:
Implemented Level-6. More code refactoring and more exceptions desired

jbkim1999 committed:
Add more exceptions, add enums, and refactor codes

jbkim1999 committed:
Add more exceptions, add enums, and refactor codes

jbkim1999 committed:
Add branch that take care of saving and loading

jbkim1999 committed:
Store task's date with LocalDate and Search date on tasks

jbkim1999 committed:
Merge branch 'branch-Level-7'
Our tasks are not saved permanently and they reset everytime when a program starts fresh.

The client should be able to save the task for future reference.

Let's use File and FileWriter to load and save the tasks as a txt file.

Txt file is light to handle and can do the job that we want to do.

jbkim1999 committed:
Handle merge conflict

jbkim1999 committed:
Add more OOP

jbkim1999 committed:
Declare packages

jbkim1999 committed:
Add Junit tests

jbkim1999 committed:
Add JavaDoc

jbkim1999 committed:
Merge branch 'branch-A-JavaDoc'

jbkim1999 committed:
Merge branch 'master' into branch-Level-9

jbkim1999 committed:
Merge branch 'master' into branch-A-CodingStandard

jbkim1999 committed:
Simulates a change done in this branch

jbkim1999 committed:
Simulates a change done in this branch

jbkim1999 committed:
Merge branch 'branch-Level-9'

jbkim1999 committed:
Simulates another change done in this branch

jbkim1999 committed:
Merge branch 'branch-A-CodingStandard'

seanleong339 commented on src/main/java/istjbot/command/DateCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.format.DateTimeParseException;
+
+public class DateCommand extends Command {
```
Perhaps add a header comment for this class?

I noticed this issue in some of the other command files too.

seanleong339 commented on src/main/java/istjbot/command/FindCommand.java:
```diff
@@ -0,0 +1,40 @@
+        String[] commandInfo = this.getFullCommand().split(" ");
+        StringBuilder keyword = new StringBuilder();
+
+        // Exception handling
```
I like how you added comments within your code to explain the logic 👍

seanleong339 commented on src/main/java/istjbot/command/ListCommand.java:
```diff
@@ -0,0 +1,29 @@
+        return false;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws BotException {
```
Perhaps add a comment explaining the execute method for each Command class?
I noticed this in other Command classes as well

pnutzz-0207 commented on src/main/java/istjbot/command/ModifyCommand.java:
```diff
@@ -0,0 +1,57 @@
+        }
+
+        switch (this.getCommandEnum()) {
+        case MARK:
```
It is great to see that you did not make the common mistake! (indentation before case) :)

pnutzz-0207 commented on src/main/java/istjbot/parser/Parser.java:
```diff
@@ -0,0 +1,55 @@
+package istjbot.parser;
+
+import istjbot.command.AddCommand;
+import istjbot.exception.BotException;
```
is this import statement not in alphabetical order?

pnutzz-0207 commented on src/main/java/istjbot/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+package istjbot.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
```
You might want to arrange the import statements in alphabetical order. The same issue is found is some of the other files as well. :)
```suggestion
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;
import java.time.LocalDate;
```

pnutzz-0207 commented on src/main/java/istjbot/task/TaskList.java:
```diff
@@ -0,0 +1,240 @@
+     *
+     * @return The number of tasks stored.
+     */
+    public int tasksSize() {
```
Will 'taskListSize' be a better name for this method? :)

pnutzz-0207 commented on src/main/java/istjbot/ui/Ui.java:
```diff
@@ -0,0 +1,139 @@
+     * @param addedTask String of the task added.
+     */
+    public void showTaskAdded(int tasksSize, String addedTask) {
+        String messageStart = "As an IstjBot, I will add this task right now. \n";
+        // May refactor messageLast?
+        String messageLast = "Now you have " + tasksSize + " ";
```
Would 'initialMessage' and  'finalMessage' be better names for these 2 variables? :)

pnutzz-0207 reviewed (commented):
Overall, good naming for all the variables! They are clear and I can understand what they do. I've put some suggestions inside, but the original names are good as well ^^~ 💯

jbkim1999 committed:
Add JUnit testing for TaskList class

jbkim1999 committed:
Merge branch 'add-gradle-support' of https://github.com/jbkim1999/ip into branch-A-Gradle

jbkim1999 committed:
Add gradle support

jbkim1999 committed:
Add support for GUI interaction

jbkim1999 committed:
Modify implementation to improve the code quality
Our implementation does not follow some of the coding standards, such as the Single Level of Abstraction Principle (SLAP).

The reader might find our code base logically incoherent.

Let's modify and extract some of our codes.

Extracting and modifying the codes this way is preferable as it will improve the readbility of our codes.

jbkim1999 committed:
Merge pull request #1 from jbkim1999/branch-A-CodeQuality

Modify implementation to improve the code quality

jbkim1999 committed:
Add assertion statements
Our codes don't have an assertion statement.

This might lead to unexpected bugs during runtime.

Let's add assertion statements to methods that assume a certain condition at the beginning.

Adding such statements is preferable as it will explicitly tell the readers what are some fundamental assumptions made when using some methods or classes.

jbkim1999 committed:
Solve merge conflict with master branch

jbkim1999 committed:
Merge pull request #2 from jbkim1999/branch-A-Assertions

Add assertion statements

jbkim1999 committed:
Add GitHub Actions

jbkim1999 committed:
Merge pull request #3 from jbkim1999/branch-A-CI

Add GitHub Actions

jbkim1999 committed:
Place static variable definition in the right place

jbkim1999 committed:
Add Extension D-Notes

jbkim1999 committed:
Merge pull request #5 from jbkim1999/branch-BCD-Extension

Add Extension D-Notes

jbkim1999 committed:
Fix minor syntax according to the automated analysis

jbkim1999 committed:
Add User Guide for IstjBot

jbkim1999 committed:
Fix bug of displaying messages through both CLI and GUI

jbkim1999 committed:
Update README.md

jbkim1999 committed:
Update README.md
